### PR TITLE
Force-recreate for service discovery resources

### PR DESCRIPTION
**Pull request by PettitWesley**
- On create, If CLI detects that a CFN stack already exists, it tries to delete it
- For namespaces, it only does force recreation if the namespace could not be found using route53 servicediscovery APIs
- This features allows the CLI to clean up old CFN stacks from previous failed attempts
- Route53 will prevent the removal of any resource that is actually in use

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: ecs-cli/modules/cli/servicediscovery/servicediscovery_app.go [hunk 1]
```diff
@@ -201,6 +218,13 @@ func createNamespace(c *cli.Context, networkMode, serviceName, clusterName strin
 	}
 
 	namespaceStackName := cfnStackName(privateDNSNamespaceStackNameFormat, clusterName, serviceName)
+
+	// if we're in this function, then we already tried to find an existing namespace and failed to find one
+	// which means that if there's a CFN stack, we should delete it, because it must be from a previous invocation that failed
```
[the remark this answers is not in the document]
Comment (PettitWesley) [reply ecs-cli/modules/cli/servicediscovery/servicediscovery_app.go]: > Also, what is the scenario in which a previous invocation of createNamespace would have created a stack that failed?

We ran into this a lot in the bug bash, you accidentally give input which is invalid (say a VPC ID that's in the wrong region) and so the CFN create fails.

**Comment by SoManyHs** [comment]
@PettitWesley Could you update the failing tests? `TestDeleteServiceDiscoveryStackNotFoundError`

File: ecs-cli/modules/cli/servicediscovery/servicediscovery_app.go [hunk 2]
```diff
@@ -194,13 +205,22 @@ func create(c *cli.Context, networkMode, serviceName string, cfnClient cloudform
...
 func createNamespace(c *cli.Context, networkMode, serviceName, clusterName string, cfnClient cloudformation.CloudformationClient, input *utils.ServiceDiscovery) (*string, error) {
 	namespaceParams := getNamespaceCFNParams(input)
 	if err := namespaceParams.Validate(); err != nil {
 		return nil, err
 	}
 
 	namespaceStackName := cfnStackName(privateDNSNamespaceStackNameFormat, clusterName, serviceName)
+
+	if err := deleteStack(namespaceStackName, serviceName, "Private DNS Namespace", cfnClient, true, true); err != nil {
+		return nil, errors.Wrapf(err, "A Private DNS Namespace CloudFormation stack for %s already exists, failed to delete existing stack: %s", serviceName, err)
```
Comment (allisaurus) [inline ecs-cli/modules/cli/servicediscovery/servicediscovery_app.go]: Is it possible an error could be returned here that is not related to an existing stack? (e.g., network issue, throttling) If so we should make the language more generic.

Comment (PettitWesley) [reply ecs-cli/modules/cli/servicediscovery/servicediscovery_app.go]: Any error returned here would have to do with the the attempted deletion of the existing stack. With the given arguments, deleteStack will only return an error if the validateStack() call succeeds first. So if there is an error here, we know that the stack exists. Errors could be returned that have to do with throttling or network issues, but since we know that a stack exists, I think that the error message is accurate.

File: ecs-cli/modules/cli/servicediscovery/servicediscovery_app.go [hunk 3]
```diff
@@ -119,39 +119,33 @@ func update(c *cli.Context, networkMode, serviceName, clusterName string, cfnCli
...
+		err = deleteStack(namspaceStackName, projectName, "Private DNS Namespace", cfnClient, false, false)
 		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-func deleteExistingStack(stackName, projectName, resource string, cfnClient cloudformation.CloudformationClient) error {
+func deleteStack(stackName, projectName, resource string, cfnClient cloudformation.CloudformationClient, ignoreValidation, cleanUp bool) error {
```
Comment (SoManyHs) [inline ecs-cli/modules/cli/servicediscovery/servicediscovery_app.go]: I don't love requiring so many parameters to a single function -- seems like a good opportunity to consider where the separation of concerns is, e.g. if some of the business logic can be handled by caller.

File: ecs-cli/modules/cli/servicediscovery/servicediscovery_app.go (hunk 2 above)
Comment (SoManyHs) [reply ecs-cli/modules/cli/servicediscovery/servicediscovery_app.go]: While I see what you're trying to do, I agree with Allie that this could be clearer -- having to pass in an unnamed boolean that determines whether validation succeeded requires the reader to look up the implementation details of `deleteStack`(see previous comment).

Comment (allisaurus) [reply ecs-cli/modules/cli/servicediscovery/servicediscovery_app.go]: This addresses my concern re: inaccurate messaging, but I concur with @SoManyHs that understanding how we got here is more complicated than ideal (though not a blocker for me ATM)
